autoarray/plot/mat_wrap/wrap/wrap_2d.py

```diff
@@ -0,0 +1,568 @@
+         """
+
+        if len(grid_grouped) == 0:
+            return
+
+        color = itertools.cycle(self.colors)
+
+        for grid_group in grid_grouped.in_grouped_list:
+
+            plt.plot(
+                np.asarray(grid_group)[:, 1],
+                np.asarray(grid_group)[:, 0],
+                c=next(color),
+                **self.config_dict_plot,
+            )
+
+
+class VectorFieldQuiver(AbstractMatWrap2D):
+    def __init__(self, **kwargs):
+        """
+        Plots a `VectorField` data structure. A vector field is a set of 2D vectors on a grid of 2d (y,x) coordinates.
+        These are plotted as arrows representing the (y,x) components of each vector at each (y,x) coordinate of it
+        grid.
+
+        This object wraps the following Matplotlib method:
+
+        https://matplotlib.org/3.3.2/api/_as_gen/matplotlib.pyplot.quiver.html
+        """
+        super().__init__(kwargs=kwargs)
+
+    def quiver_vector_field(self, vector_field: vector_fields.VectorFieldIrregular):
+        """
+         Plot a vector field using the matplotlib method `plt.quiver` such that each vector appears as an arrow whose
+         direction depends on the y and x magnitudes of the vector.
+
+         Parameters
+         ----------
+         vector_field : VectorFieldIrregular
+             The vector field that is plotted using `plt.quiver`.
+         """
+        plt.quiver(
+            vector_field.grid[:, 1],
+            vector_field.grid[:, 0],
+            vector_field[:, 1],
+            vector_field[:, 0],
+            **self.config_dict_quiver,
+        )
+
+
+class PatchOverlay(AbstractMatWrap2D):
+    def __init__(self, **kwargs):
+        """
+        Adds patches to a plotted figure using matplotlib `patches` objects.
+
+        The coordinate system of each `Patch` uses that of the figure, which is typically set up using the plotted
+        data structure. This makes it straight forward to add patches in specific locations.
+
+        This object wraps methods described in below:
+
+        https://matplotlib.org/3.3.2/api/collections_api.html
+        """
+        super().__init__(kwargs=kwargs)
+
+    def overlay_patches(self, patches: typing.Union[ptch.Patch]):
+        """
+        Overlay a list of patches on a figure, for example an `Ellipse`.
+        `
+        Parameters
+        ----------
+        patches : [Patch]
+            The patches that are laid over the figure.
+        """
+        patch_collection = PatchCollection(
+            patches=patches, **self.config_dict_patch_collection
+        )
+
+        plt.gcf().gca().add_collection(patch_collection)
+
+
+class VoronoiDrawer(AbstractMatWrap2D):
+    def __init__(self, **kwargs):
+        """
+        Draws Voronoi pixels from a `MapperVoronoi` object (see `inversions.mapper`). This includes both drawing
+        each Voronoi cell and coloring it according to a color value.
+

```

Comment:
If you make AbstractMatWrap2D take **kwargs then this constructor does not need to exist.

You can put docs for the class directly under the class signature